mars/services/scheduling/supervisor/assigner.py

```diff
@@ -90,6 +107,8 @@ async def assign_subtasks(self, subtasks: List[Subtask]):
                         continue
                     meta = inp_metas[inp.key]
                     for band in meta['bands']:
+                        if band not in available_bands:
+                            band = await self.reassign_band()
```

Comment:
Ditto. This is becoming even more expansive as we have already fetched available bands.